pkg/thanos/rules.go

```diff
@@ -51,9 +51,34 @@ func (o *Operator) createOrUpdateRuleConfigMaps(ctx context.Context, t *monitori
 		return nil, err
 	}
 
-	newRules, err := o.selectRules(t, namespaces)
+	excludedFromEnforcement := t.Spec.ExcludedFromEnforcement
+	// append the deprecated PrometheusRulesExcludedFromEnforce
+	for _, rule := range t.Spec.PrometheusRulesExcludedFromEnforce {
+		excludedFromEnforcement = append(excludedFromEnforcement,
+			monitoringv1.ObjectReference{
+				Namespace: rule.RuleNamespace,
+				Group:     monitoring.GroupName,
+				Resource:  monitoringv1.PrometheusRuleName,
+				Name:      rule.RuleName,
+			})
+	}
```

Comment:
This would be true for Prometheus and false for Thanos right? 

```
nsLabeler := namespacelabeler.New(
		t.Spec.EnforcedNamespaceLabel,
		append(t.Spec.ExcludedFromEnforcement, monitoringv1.ToObjectReferences(t.Spec.PrometheusRulesExcludedFromEnforce)...)
		false,
	)
```

Reply:
yes correct, sorry for the confusion.

Reply:
Are we good with the changes?